core/client/fs/src/main/java/alluxio/client/file/cache/store/PageStoreDir.java

```diff
@@ -0,0 +1,168 @@
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.client.file.cache.store;
+
+import static com.google.common.base.Preconditions.checkState;
+
+import alluxio.client.file.cache.PageInfo;
+import alluxio.client.file.cache.PageStore;
+import alluxio.client.file.cache.evictor.CacheEvictor;
+import alluxio.conf.AlluxioConfiguration;
+
+import com.google.common.collect.ImmutableList;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.function.Consumer;
+
+/**
+ * Directory of page store.
+ */
+public interface PageStoreDir {
+  /**
+   * Create a list of PageStoreDir based on the configuration.
+   * @param conf AlluxioConfiguration
+   * @return A list of LocalCacheDir
+   * @throws IOException
+   */
+  static List<PageStoreDir> createPageStoreDirs(AlluxioConfiguration conf)
+      throws IOException {
+    return PageStoreOptions.create(conf).stream().map(options -> createPageStoreDir(conf, options))
+        .collect(ImmutableList.toImmutableList());
+  }
+
+  /**
+   * Create an instance of PageStoreDir.
+   *
+   * @param conf
+   * @param pageStoreOptions
+   * @return PageStoreDir
+   */
+  static PageStoreDir createPageStoreDir(AlluxioConfiguration conf,
+                                         PageStoreOptions pageStoreOptions) {
+    switch (pageStoreOptions.getType()) {
+      case LOCAL:
+        checkState(pageStoreOptions instanceof LocalPageStoreOptions);
+        return new LocalPageStoreDir(
+            (LocalPageStoreOptions) pageStoreOptions,
+            PageStore.openOrCreatePageStore(pageStoreOptions),
+            CacheEvictor.create(conf)
+        );
+      case ROCKS:
+        return new RocksPageStoreDir(
+            pageStoreOptions,
+            PageStore.openOrCreatePageStore(pageStoreOptions),
+            CacheEvictor.create(conf)
+        );
+      case MEM:
+        return new MemoryPageStoreDir(
+            pageStoreOptions,
+            (MemoryPageStore) PageStore.openOrCreatePageStore(pageStoreOptions),
+            CacheEvictor.create(conf)
+        );
+      default:
+        throw new IllegalArgumentException(String.format("Unrecognized store type %s",
+            pageStoreOptions.getType().name()));
+    }
+  }
+
+  /**
+   * @param fileBuckets number of buckets
+   * @param fileId file id
+   * @return file bucket
+   */
+  static String getFileBucket(int fileBuckets, String fileId) {
+    return Integer.toString(Math.floorMod(fileId.hashCode(), fileBuckets));
+  }
+
+  /**
+   * @return root path
+   */
+  Path getRootPath();
+
+  /**
+   * Getter for pageStore.
+   * @return pageStore
+   */
+  PageStore getPageStore();
+
+  /**
+   * Getter for capacity.
+   * @return capacity
+   */
+  long getCapacity();
+
+  /**
+   * Reset page store.
+   */
+  void reset();
+
+  /**
+   * Traverse the pages under this dir.
```

Comment:
doesn't make sense to me.

Reply:
this one is a page scanner actually, I changed the method name to scanPages and also updated the java doc,  does that makes more sense to you?